File: CSharp/ValidParens.cs
public bool IsValid(string s) {
        char[] chars = s.ToCharArray();
        var myStack = new Stack<char>();
        HashSet<char> openBrackets = new HashSet<char> { '{', '(', '[' };
        foreach(char c in chars)
        {
            if (openBrackets.Contains(c))
            {
                myStack.Push(c);
            }
            else
            {
                if (myStack.Count == 0)
                {
                    return false;
                }
                var current = (char)myStack.Pop();
                if (current == '{' && c != '}') return false;
                if (current == '(' && c != ')') return false;
                if (current == '[' && c != ']') return false;
            }
        }
        return myStack.Count == 0;
}

File: java/ValidParens.java
public boolean matches(char a, char b) {
  return (a == '{' && b == '}') || (a == '(' && b == ')') || (a == '[' && b == ']');
}

public boolean isValid(String s) {
  Stack<Character> stack = new Stack<>();
  for (char c : s.toCharArray()) {
    if (c == '{' || c == '(' || c == '[') {
        stack.push(c);
    } else {
        if (stack.isEmpty()) return false;
        char last = stack.pop();
        if (!matches(last, c)) return false;
    }
  }
  return stack.isEmpty();
}
